aten/src/ATen/native/cpu/MaxPooling.cpp

```diff
@@ -0,0 +1,57 @@
+#include <ATen/ATen.h>
+#include <ATen/Parallel.h>
+#include <ATen/cpu/vec256/vec256.h>
+#include <ATen/native/MaxPooling.h>
+
+namespace at {
+namespace native {
+
+namespace {
+
+template <typename scalar_t>
+inline void max_pool1d_kernel(
+    scalar_t* C10_RESTRICT op,
+    const scalar_t* C10_RESTRICT ip,
+    const PoolingParams1D& p) {
+  for (int64_t kj = 0; kj < p.KW; ++kj) {
```

Comment:
Did you check that this actually benefits from auto vectorization? If it does not, it should not be in the cpu folder.

Reply:
Yes I did, it does benefit especially when using C10_RESTRICT which can only be used in cpu folder.